### PR TITLE
[IN-89] Redirect Create page to View page when creating JobPosting on last step

### PR DESCRIPTION
**Pull request by tommytwm**
#### User Story
[[IN-89] Redirect Create page to View page when creating JobPosting on last step](https://chaosneutral.atlassian.net/jira/software/projects/IN/boards/1?selectedIssue=IN-89)

#### Changes made
- Added redirect in `Review` page to go to `View` page when job posting is added
- Added new `User` typed redux action to update the user store with the new job posting to fix overwriting current user store
- Added bulk endpoint call `/api/jobs/bulk` that takes in an array of `jobId`s and returns all the job postings from the database
- Added new `jobId` field to `TableBasic` so that the table in the `Profile` page is able to redirect to the `View` page
- Added new `dateCreated` field in `JobPostingSchema` to keep track of when the job posting was created

#### Does your new code introduce new warnings on dev console?
It shouldn't have. It was fine on my machine but I'm getting inconsistent results from a variety of scenarios. Double check on your machine.

#### Test Steps
1. Login
2. Create a job posting
3. Once you've reached the `Review` step, click 'Create' to add the job posting
4. You should be redirected to the respective `View` page with URI `/view/:jobId`
5. Click on the back button on the top left of the `View` page and you should be redirected to the `Profile` page
6. You should see the newly updated list of jobs in the table on the right of the `Profile` page

### CODE REVIEW

File: internify/backend/routes/api/jobs.js [hunk 1]
```diff
@@ -1,55 +1,89 @@
...
+  Promise.all(promises)
+    .then((values) => {
+      jobPostings.push(values);
+    })
+    .catch((err) => {
+      res.status(404).json({ success: false });
+    })
+    .finally(() => {
+      res.status(200).json(jobPostings);
+    });
```
Comment (tommytwm) [inline internify/backend/routes/api/jobs.js:43]: The freaking 310 trauma that erected when I wrote this.

File: internify/backend/routes/api/jobs.js [hunk 2]
```diff
@@ -1,55 +1,89 @@
-var express = require('express');
+var express = require("express");
 var router = express.Router();
-var JobPostingData = require('../../models/JobPostingData');
+var JobPostingData = require("../../models/JobPostingData");
 
 //@route    GET api/jobs/getAll
```
Comment (JudeShamsi) [inline internify/backend/routes/api/jobs.js]: update to match actual route -> api/jobs/

File: internify/backend/routes/api/jobs.js [hunk 3]
```diff
@@ -1,55 +1,89 @@
...
-router.get('/', function(req, res, next) {
+router.get("/", function (req, res, next) {
   JobPostingData.find()
-    .then(jobs => res.status(200).json(jobs))
-    .catch(err => res.status(404).json({success: false}));
+    .then((jobs) => res.status(200).json(jobs))
+    .catch((err) => res.status(404).json({ success: false }));
+});
+
+//@route    GET api/jobs/getBulk
```
Comment (JudeShamsi) [inline internify/backend/routes/api/jobs.js]: update to api/jobs/bulk

File: internify/backend/routes/api/jobs.js (hunk 1 above)
Comment (JudeShamsi) [reply internify/backend/routes/api/jobs.js:43]: beautiful though <3

File: internify/backend/routes/api/jobs.js [hunk 4]
```diff
@@ -1,55 +1,89 @@
...
+    })
+    .catch((err) => {
+      res.status(404).json({ success: false });
+    })
+    .finally(() => {
+      res.status(200).json(jobPostings);
+    });
 });
 
 //@route    GET api/jobs/getOne/:id
```
Comment (JudeShamsi) [inline internify/backend/routes/api/jobs.js]: update to api/jobs/:id

File: internify/src/components/molecules/JobPosting.js [hunk 5]
```diff
@@ -27,10 +31,12 @@ const JobPosting = (props) => {
     details.candidates,
   ];
 
-  function sendJob(){
-    props.actions.addNewJob(data);
+  function sendJob() {
+    props.actions.addJob({ ...data, dateCreated: Date.now() });
+    props.actions.updateJobsOfUser({ authId: user.uid, jobPostings: [jobId] });
```
Comment (JudeShamsi) [inline internify/src/components/molecules/JobPosting.js:36]: is this going to re-write the jobId, should we not add a spread operator?

**Comment by ReneReid** [comment]
Hey @tommytwm  - great work here! love the functionality! Just a couple things -  in reference to the **test steps 5, 6** above, when I go back from the newly created job to my profile page, i actually do not see a list of the created jobs.  However, I do see all the jobs if I logout and login again.  That might be something more related to the **profile page** than the job creation stages though. 

The second thing is an error message - which I only saw once and have not yet been able to reproduce - when at the last step of creating a job i.e. hitting the "Create" button.  The error message "TypeError: Cannot read property 'uid' of null" came up.  I've attempted to recreate conditions to see how consistent this error message is, but without success so far.  I guess its something we'll have to be aware of while testing. 

(The following issue occurs in other branches as well) With regards to the review page, I think there is either an additional bullet point or there is a field missing in the content (please see attached picture for ease of reference).

Again, many thanks for this, Tommy!

Screen Shot 2021-07-26 at 12 51 19 PM

File: internify/src/components/molecules/JobPosting.js (hunk 5 above)
Comment (tommytwm) [reply internify/src/components/molecules/JobPosting.js:36]: I agree, let's look into this later today

**Review by ghost — CHANGES_REQUESTED** [review]
Since both Review Page and View Page involve copying some stuff that is displayed (JobPosting vs ViewPosting, the following errors exist for both of them).

1. each child in a list should be have the unique key prop (looks like an li issue, may need to add a key variable to the li and it should be fixed)
2. I notice that on technical requirements, the highest degree is shown as a requirement in review (should be the minimal)
3. I just notice a 404 Not Found on the Put request in console (may have been there previously)

Otherwise, no issues that I can find.

File: internify/src/components/molecules/JobPosting.js (hunk 5 above)
Comment (tommytwm) [reply internify/src/components/molecules/JobPosting.js:36]: This is not going to re-write the jobId. The new user action pushes the new jobId to the User store array of job postings

**Comment by tommytwm** [comment]
@aleemer 
> Since both Review Page and View Page involve copying some stuff that is displayed (JobPosting vs ViewPosting, the following errors exist for both of them).
> 
>     1. each child in a list should be have the unique key prop (looks like an li issue, may need to add a key variable to the li and it should be fixed)
> 
>     2. I notice that on technical requirements, the highest degree is shown as a requirement in review (should be the minimal)
> 
>     3. I just notice a 404 Not Found on the Put request in console (may have been there previously)
> 
> 
> Otherwise, no issues that I can find.

(1) ~~This error pops up when and if the user copy pastes any of the information from prior fields in the Create section to any of the continuing fields. For testing purposes, this can be ignored for now. However, it is assumed that users will not be copying and pasting the same content on any of the fields in Create~~ Fixed
(2) ~~I'm not sure I understand this. Could you clarify?~~ Fixed
(3) I'm not able to replicate this. Would you be able to provide steps?

**Comment by ReneReid** [comment]
@JudeShamsi @tommytwm  - this is great!! thank you for getting it up and running!  I've played around with the create/view pages and everything is as expected for those functions. I can create a job post, go back and see it included in my Profile page, view it, go back to my profile page etc.  
The next story im working on - Editing a job post - will very likely be building on this. Excited!

**Review by ReneReid — APPROVED** [review]

**Review by ghost — APPROVED** [review]
The put 404 thing may just be an issue with refreshing too many times.

**Pull request merged**